chain-tx-enclave-next/enclave-ra/ra-client/src/verifier.rs

```diff
@@ -339,3 +377,233 @@ pub struct CertVerifyResult {
+            public_key,
+            Utc::now(),
+        );
+
+        assert!(matches!(
+            result.unwrap_err(),
+            EnclaveCertVerifierError::AttestationReportSigningCertificateVerificationError(_)
+        ));
+    }
+
+    #[test]
+    fn test_verify_attestation_report_missing_attestation_report_signing_certificate() {
+        let ias_ca = include_bytes!("../test/Intel_SGX_Attestation_RootCA.pem");
+
+        let attestation_report = include_bytes!("../test/valid_attestation_report.json");
+        let mut attestation_report: AttestationReport =
+            serde_json::from_slice(&attestation_report[..]).unwrap();
+        attestation_report.signing_cert = Vec::new();
+        let attestation_report = serde_json::to_vec(&attestation_report).unwrap();
+
+        let report_data = base64::decode("1g+Nvsow2LXbrJVq/8YS5wMUd+GTeOkBegUmnGtcfyLSS0qP6ufwO2HEDV70O4W/tFDx57tziaOWd6OJjenAeg==").unwrap();
+        let public_key = &[&[4], report_data.as_slice()].concat();
+
+        let verifier_config = EnclaveCertVerifierConfig {
+            signing_ca_cert_pem: ias_ca.to_vec().into(),
+            valid_enclave_quote_statuses: vec![
+                "OK".into(),
+                "CONFIGURATION_AND_SW_HARDENING_NEEDED".into(),
+            ]
+            .into(),
+            report_validity_secs: 86400,
+            enclave_info: None,
+        };
+        let verifier = EnclaveCertVerifier::new(verifier_config).unwrap();
+        let result = verifier.verify_attestation_report(
+            attestation_report.as_slice(),
+            public_key,
+            Utc::now(),
+        );
+
+        assert!(matches!(
+            result.unwrap_err(),
+            EnclaveCertVerifierError::MissingAttestationReportSigningCertificate
+        ));
+    }
+
+    #[test]
+    fn test_verify_attestation_report_public_key_mismatch() {
+        let ias_ca = include_bytes!("../test/Intel_SGX_Attestation_RootCA.pem");
+        let attestation_report = include_bytes!("../test/valid_attestation_report.json");
+        let report_data = base64::decode("1g+Nvsow2LXbrJVq/8YS5wMUd+GTeOkBegUmnGtcfyLSS0qP6ufwO2HEDV70O4W/tFDx67tziaOWd6OJjenAeg==").unwrap();
+        let public_key = &[&[4], report_data.as_slice()].concat();
+
+        let verifier_config = EnclaveCertVerifierConfig {
+            signing_ca_cert_pem: ias_ca.to_vec().into(),
+            valid_enclave_quote_statuses: vec![
+                "OK".into(),
+                "CONFIGURATION_AND_SW_HARDENING_NEEDED".into(),
+            ]
+            .into(),
+            report_validity_secs: 86400,
+            enclave_info: None,
+        };
+        let verifier = EnclaveCertVerifier::new(verifier_config).unwrap();
+        let result = verifier.verify_attestation_report(attestation_report, public_key, Utc::now());
+
+        assert!(matches!(
+            result.unwrap_err(),
+            EnclaveCertVerifierError::PublicKeyMismatch
+        ));
+    }
+}
```

Comment:
all the tests use `Utc::now()` which would make them non-deterministic (e.g. valid report may become invalid after this expires) -- it may be better to use some fixed reference time in tests

Reply:
but not sure if this will still be an issue -- e.g. `webpki` methods may still use the current system time?

Reply:
found the issue as well
fixed in previous commit